src/jit/importer.cpp

```diff
@@ -11801,9 +11801,10 @@ void Compiler::impImportBlockCode(BasicBlock* block)
 
                 // Check for assignment to same array, ie. arrLcl[i] = arrLcl[j]
                 // This does not need CORINFO_HELP_ARRADDR_ST
-                if (arrayNodeFrom->OperGet() == GT_INDEX && arrayNodeFrom->gtOp.gtOp1->gtOper == GT_LCL_VAR &&
+                if (arrayNodeFrom->OperGet() == GT_INDEX && arrayNodeFrom->AsOp()->gtOp1->gtOper == GT_LCL_VAR &&
```

Comment:
That condition is so unreadable (with or without this change), another note.

Reply:
Ah, a fine example of what I was talking about earlier. This casts to `GenTreeOp*` rather than using `gtGetOp1()` directly but still gets it wrong - `AsIndex()->Arr()` would be more readable.